core/modules/replica/schema/migrate-0-to-1.sql

```diff
@@ -0,0 +1,509 @@
+ worker services';
+
+
+CREATE TABLE IF NOT EXISTS `config_database_family` (
+
+  `name`                   VARCHAR(255)  NOT NULL ,
+  `min_replication_level`  INT UNSIGNED  NOT NULL ,    -- minimum number of replicas per chunk
+  `num_stripes`            INT UNSIGNED  NOT NULL ,
+  `num_sub_stripes`        INT UNSIGNED  NOT NULL ,
+  `overlap`                DOUBLE        NOT NULL ,
+
+  PRIMARY KEY (`name`)
+)
+ENGINE = InnoDB
+COMMENT = 'Groups of databases which require coordinated replication efforts';
+
+
+CREATE TABLE IF NOT EXISTS `config_database` (
+
+  `database`     VARCHAR(255)  NOT NULL ,
+  `family_name`  VARCHAR(255)  NOT NULL ,
+
+  `is_published` BOOLEAN DEFAULT TRUE ,
+
+  `chunk_id_key`     VARCHAR(255)  DEFAULT "" ,
+  `sub_chunk_id_key` VARCHAR(255)  DEFAULT "" ,
+
+  -- Each database is allowed to belong to one family only
+  --
+  PRIMARY KEY (`database`) ,
+  UNIQUE  KEY (`database`,`family_name`) ,
+
+  CONSTRAINT `config_database_fk_1`
+    FOREIGN KEY (`family_name` )
+    REFERENCES `config_database_family` (`name` )
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+)
+ENGINE = InnoDB
+COMMENT = 'Databases managed by the replication system';
+
+
+CREATE TABLE IF NOT EXISTS `config_database_table` (
+
+  `database`  VARCHAR(255)  NOT NULL ,
+  `table`     VARCHAR(255)  NOT NULL ,
+
+  `is_partitioned` BOOLEAN NOT NULL ,
+  `is_director`    BOOLEAN NOT NULL ,
+
+  `director_key`  VARCHAR(255) DEFAULT "" ,
+  `latitude_key`  VARCHAR(255) DEFAULT "" , -- Name for latitude (declination) column in this table
+  `longitude_key` VARCHAR(255) DEFAULT "" , -- Name for longitude (right ascension) column in this table
+
+  PRIMARY KEY (`database`, `table`) ,
+
+  CONSTRAINT `config_database_table_fk_1`
+    FOREIGN KEY (`database` )
+    REFERENCES `config_database` (`database` )
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+)
+ENGINE = InnoDB
+COMMENT = 'Database tables managed by the replication system';
+
+
+CREATE TABLE IF NOT EXISTS `config_database_table_schema` (
+
+  `database`  VARCHAR(255)  NOT NULL ,
+  `table`     VARCHAR(255)  NOT NULL ,
+
+  `col_position` INT NOT NULL ,             -- for preserving an order of columns in the table
+  `col_name`     VARCHAR(255)  NOT NULL ,
+  `col_type`     VARCHAR(255)  NOT NULL ,
+
+  UNIQUE KEY (`database`, `table`, `col_position`) ,
+  UNIQUE KEY (`database`, `table`, `col_name`) ,
+
+  CONSTRAINT `config_database_table_schema_fk_1`
+    FOREIGN KEY (`database`, `table`)
+    REFERENCES `config_database_table` (`database`, `table`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+)
+ENGINE = InnoDB
+COMMENT = 'Table schemas for the managed tables. The schemas are registered during
+  catalog ingests, and they are used by the Ingest system.';
+
+
+CREATE TABLE IF NOT EXISTS `controller` (
+
+  `id`  VARCHAR(255) NOT NULL ,
+
+  `hostname`  VARCHAR(255) NOT NULL ,
+  `pid`       INT          NOT NULL ,
+
+  `start_time`  BIGINT UNSIGNED NOT NULL ,
+
+  PRIMARY KEY (`id`)
```

Comment:
I don't think the extra empty lines are helpful in these create table statements.